### ISSUE TITLE
Matplotlib import error.

### ISSUE THREAD

**Issue by conor-byrne** [ContinuumIO/anaconda-issues#511]
I'm having the same issue as described here:

http://stackoverflow.com/questions/33459574/import-matplotlib-png-as-png-import-error-dll-load-failed-the-specified-mod

The answer states that it's a very recent issue and due to a scipy compile error. I've no idea where he got this info but I thought it'd be good to pass it on.

As for me, I ran conda install anaconda but no luck. Still getting the same error. Running import matplotlib.pyplot as plt in ipython gives:

ImportError                               Traceback (most recent call last)
<ipython-input-1-eff513f636fd> in <module>()
----> 1 import matplotlib.pyplot as plt

C:\Users\Conor\Anaconda3\lib\site-packages\matplotlib\pyplot.py in <module>()
     25
     26 import matplotlib
---> 27 import matplotlib.colorbar
     28 from matplotlib import style
     29 from matplotlib import _pylab_helpers, interactive

C:\Users\Conor\Anaconda3\lib\site-packages\matplotlib\colorbar.py in <module>()
     32 import matplotlib.artist as martist
     33 import matplotlib.cbook as cbook
---> 34 import matplotlib.collections as collections
     35 import matplotlib.colors as colors
     36 import matplotlib.contour as contour

C:\Users\Conor\Anaconda3\lib\site-packages\matplotlib\collections.py in <module>()
     25 import matplotlib.artist as artist
     26 from matplotlib.artist import allow_rasterization
---> 27 import matplotlib.backend_bases as backend_bases
     28 import matplotlib.path as mpath
     29 from matplotlib import _path

C:\Users\Conor\Anaconda3\lib\site-packages\matplotlib\backend_bases.py in <module>()
     54
     55 import matplotlib.tight_bbox as tight_bbox
---> 56 import matplotlib.textpath as textpath
     57 from matplotlib.path import Path
     58 from matplotlib.cbook import mplDeprecation

C:\Users\Conor\Anaconda3\lib\site-packages\matplotlib\textpath.py in <module>()
     20 from matplotlib.ft2font import FT2Font, KERNING_DEFAULT, LOAD_NO_HINTING
     21 from matplotlib.ft2font import LOAD_TARGET_LIGHT
---> 22 from matplotlib.mathtext import MathTextParser
     23 import matplotlib.dviread as dviread
     24 from matplotlib.font_manager import FontProperties

C:\Users\Conor\Anaconda3\lib\site-packages\matplotlib\mathtext.py in <module>()
     61
     62 import matplotlib.colors as mcolors
---> 63 import matplotlib._png as _png
     64 ####################
     65

ImportError: DLL load failed: The specified module could not be found.

**Comment by ale-jin** [comment]
Same problem here, I tried the following steps without success:
* Clean install of miniconda 
* Clean install of Anaconda
* Install MS DLL's (http://stackoverflow.com/questions/1052831/matplotlib-dll-load-failed-when-import-pylab)

**Comment by Eric89GXL** [comment]
Confirming this bug, breaks AppVeyor CI builds on Windows 64-bit, at least for Python2.7. See AppVeyor log:

https://ci.appveyor.com/project/Eric89GXL/mne-python/build/1.0.3765

**Comment by msarahan** [comment]
duplicate of https://github.com/conda/conda/issues/1753

Please follow instructions there.  I think you need to add (PREFIX)\Library\bin to PATH somehow.  The miniconda installer does not do this for the root environment.  It will in future releases, but old ones do not.

Workaround for CI is to create and activate an environment instead of using the root env (@Eric89GXL, let me know if you want a PR): https://ci.appveyor.com/project/msarahan/mne-python/build/1.0.2

For local installations, you can add (your prefix)\Library\bin to PATH yourself, manually, if you want to use the root env, or create env's and activate them.  The activate script sets correct PATH, but only for that console session.

See also: https://github.com/ContinuumIO/anaconda-issues/issues/509 - we are working on a fix that should remedy this issue for older Miniconda installers transparently.

Closing this issue to consolidate discussion.  Please continue discussion at https://github.com/conda/conda/issues/1753

**Comment by Eric89GXL** [comment]
@msarahan thanks for the fix!

**Comment by msarahan** [comment]
This fix has been released as conda-env-2.4.4-1.  It has been tested to work with both user-level, and system-level installations of Miniconda version 3.16 and Anaconda 2.3.  Updating from earlier versions should behave similarly.

You will see a message like:
```
"Ensuring that C:\Users\builder\Miniconda\Library\bin is on user PATH environment variable."

SUCCESS: Specified value was saved.
Please open a new console window to make sure that the new PATH is in effect.
```

when the script runs.  You can view the script in (your prefix)\Scripts\.conda-env-post-link.bat
